pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -370,6 +372,48 @@ func (l *listUsersQuery) expandIntersection(
 	return <-errChan
 }
 
+func (l *listUsersQuery) expandExclusion(
+	ctx context.Context,
+	req listUsersRequest,
+	rewrite *openfgav1.Userset_Difference,
+	foundUsersChan chan<- *openfgav1.User,
+) error {
+	baseFoundUsersCh := make(chan *openfgav1.User, 1)
+	subtractFoundUsersCh := make(chan *openfgav1.User, 1)
+
+	errChan := make(chan error, 2)
+	go func() {
+		errChan <- l.expandRewrite(ctx, req, rewrite.Difference.GetBase(), baseFoundUsersCh)
+		close(baseFoundUsersCh)
+
+		errChan <- l.expandRewrite(ctx, req, rewrite.Difference.GetSubtract(), subtractFoundUsersCh)
+		close(subtractFoundUsersCh)
+	}()
+
+	baseFoundUsersCountMap := make(map[string]struct{}, 0)
+	for fu := range baseFoundUsersCh {
+		key := tuple.UserProtoToString(fu)
+		baseFoundUsersCountMap[key] = struct{}{}
+	}
+	subtractFoundUsersCountMap := make(map[string]struct{}, len(baseFoundUsersCountMap))
+	for fu := range subtractFoundUsersCh {
+		key := tuple.UserProtoToString(fu)
+		subtractFoundUsersCountMap[key] = struct{}{}
+	}
+
+	for key := range baseFoundUsersCountMap {
+		if _, isSubtracted := subtractFoundUsersCountMap[key]; !isSubtracted {
+			// Iterate over base users because at minimum they need to pass
+			// but then they are further compared to the subtracted users map.
+			// If users exist in both maps, they are excluded. Only users that exist
+			// solely in the base map will be returned.
+			foundUsersChan <- tuple.StringToUserProto(key)
+		}
+	}
+
+	return <-errChan
```

Comment:
This will only return the last error.

i think we should do

```go
var errs *multierror.Error
	go func() {
		err1 := l.expandRewrite(ctx, req, rewrite.Difference.GetBase(), baseFoundUsersCh)
                 if err1 != nil {
			multierror.Append(errs, err1)
	        }
		close(baseFoundUsersCh)

		err2 := l.expandRewrite(ctx, req, rewrite.Difference.GetSubtract(), subtractFoundUsersCh)
                if err2 != nil {
			multierror.Append(errs, err2)
	        }
		close(subtractFoundUsersCh)
	}()

...
return errs
```